src/meltano/core/hub/client.py

```diff
@@ -108,15 +108,29 @@ def __init__(self, project: Project) -> None:
 
             self.session.headers["X-Project-ID"] = project_id
 
+        if self.hub_url_auth:
+            self.session.headers.update({"Authorization": self.hub_url_auth})
+
     @property
     def hub_api_url(self):
         """Return the URL of the Hub API.
 
         Returns:
             The URL of the Hub API.
         """
+        hub_api_root = self.settings_service.get("hub_api_root")
         hub_url = self.settings_service.get("hub_url")
-        return f"{hub_url}/meltano/api/v1"
+
+        return hub_api_root or f"{hub_url}/meltano/api/v1"
```

Comment:
> I'm assuming a logic expecting "one or none but not both". And "if both are passed, `hub_api_root` wins and the other is ignored".

@aaronsteers I think this implementation meets those requirements. Does this have your 👍?

Reply:
@edgarrmondragon -  Looks good! Thanks!